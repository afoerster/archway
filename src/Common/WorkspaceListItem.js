import React from 'react';
import PropTypes from 'prop-types';
import { Card, List, Avatar } from 'antd';

import Color from './Colors';

const WorkspaceStatus = () => (
  <div
    style={{
      width: '50%',
      backgroundColor: 'green',
      display: 'inline-block',
      color: 'white',
      marginTop: 5
    }}>
    Approved
  </div>
);

const detailStyle = {
  width: '25%',
  textAlign: 'center',
  boxShadow: 'none'
};

const WorkspaceDetail = ({ label, value }) => (
  <Card.Grid style={detailStyle}>
    <h2 style={{ color: Color.Orange.rgb().string(), fontWeight: 100, textAlign: 'center' }}>
      {value}
    </h2>
    <h5 style={{ textAlign: 'center' }}>
      {label}
    </h5>
  </Card.Grid>
)

const WorkspaceListItem = ({ workspace, onSelected }) => {
  const gridStyle = { width: '33%', textAlign: 'center' };
  const {
    id,
    name,
  } = workspace;
  return (
    <List.Item>
      <Card
        style={{ textAlign: 'center' }}
        onClick={() => onSelected(`/workspaces/${id}`)}
        hoverable>
        <Avatar icon="user" />
        <h2 style={{ textAlign: 'center' }}>{name}</h2>
        <h4>partially approved</h4>
        <WorkspaceDetail label="DBs" value={1} />
        <WorkspaceDetail label="Pools" value={1} />
        <WorkspaceDetail label="Topics" value={0} />
        <WorkspaceDetail label="Apps" value={1} />
      </Card>
    </List.Item>
  );
};

WorkspaceListItem.propTypes = {
  item: PropTypes.shape({
    id: PropTypes.number.isRequired,
    name: PropTypes.string.isRequired,
    requester: PropTypes.string.isRequired,
    requested_date: PropTypes.string.isRequired,
    single_user: PropTypes.bool.isRequired,
  }).isRequired,
  onSelected: PropTypes.func.isRequired,
};

export default WorkspaceListItem;