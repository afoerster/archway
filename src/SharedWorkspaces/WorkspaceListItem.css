.WorkspaceListItem {
    position: relative;
    cursor: pointer;
    padding: 30px 75px 30px 50px;
    box-shadow: 0 3px 6px rgba(0, 0, 0, .16);
    background-color: white;
    border-radius: 10px;
    display: flex;
    flex-direction: column;
    align-items: center;
    margin: 10px;
    justify-content: center;
    height: 150px;
    width: 300px;
}

.WorkspaceListItem > .fa-arrow-circle-right {
    position: absolute;
    top: 44%;
    right: 25px;
    color: lightgray;
    font-size: 32px;
}

.WorkspaceListItem-title {
    text-transform: uppercase;
}

.WorkspaceListItem-created {
    text-align: center;
    text-transform: uppercase;
    font-weight: 300;
    font-size: 12px;
}