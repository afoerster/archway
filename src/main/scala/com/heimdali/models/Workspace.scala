package com.heimdali.models

import com.typesafe.config.Config

import scala.collection.immutable.Queue

import scala.collection.JavaConverters._

trait Workspace[T] {
  def configName: String

  def workspaceId: T

  def poolName: String

  def requestedCores(configuration: Config): Int =
    configuration.getInt(s"workspaces.$configName.defaultCores")

  def requestedMemory(configuration: Config): Int =
    configuration.getInt(s"workspaces.$configName.defaultMemory")

  def requestedDiskSize(configuration: Config): Int =
    configuration.getInt(s"workspaces.$configName.defaultSize")

  def parentPools(configuration: Config): Queue[String] =
    Queue(configuration.getString(s"workspaces.$configName.poolParents").split(",") :_*)

  def databaseName: String

  def role(configuration: Config): String

  def dataDirectory(configuration: Config): String

  def groupName(configuration: Config): String

  def initialMembers: Seq[String]

  def onBehalfOf: Option[String]
}
