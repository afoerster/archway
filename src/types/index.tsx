export interface Login {
    email: String
    password: String
}

export interface Auth {
    token: String
    error: String
    loading?: Boolean
    loggingIn: Boolean
}

export interface StoreState {
    auth: Auth
}