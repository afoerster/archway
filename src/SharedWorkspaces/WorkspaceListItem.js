import React from "react";
import TimeAgo from 'timeago-react';
import "./WorkspaceListItem.css";

const WorkspaceListItem = ({push, workspace: {id, name, created, created_by}}) => (
    <div onClick={() => push("/workspace/" + id + "/overview")} className="WorkspaceListItem">
        <i className="fas fa-arrow-circle-right"></i>
        <h1 className="WorkspaceListItem-title">{name}</h1>
        <div className="WorkspaceListItem-created">
            Created <TimeAgo datetime={created}/><br/>by {created_by}
        </div>
    </div>
);

export default WorkspaceListItem;