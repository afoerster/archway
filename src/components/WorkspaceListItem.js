import React from "react";
import TimeAgo from 'timeago-react';
import "./WorkspaceListItem.css";

const WorkspaceListItem = ({workspace: {id, name, created_by, created}}) => {
    return (
        <div key={id} className="WorkspaceListItem">
            <i className="fas fa-arrow-circle-right"></i>
            <h1 className="WorkspaceListItem-title">{name}</h1>
            <div className="WorkspaceListItem-created">
                Created <TimeAgo datetime={created} /><br />by {created_by}
            </div>
        </div>
    );
};

export default WorkspaceListItem;