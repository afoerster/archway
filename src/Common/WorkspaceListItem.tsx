import * as React from 'react';
import { Card, List, Icon, Tag } from 'antd';

import Color from './Colors';
import { Workspace } from '../WorkspaceListing/Workspace';

interface DetailProps {
  label: string
  value: any
}

const WorkspaceDetail = ({ label, value }: DetailProps) => (
  <Card.Grid style={{ width: '25%', textAlign: 'center', boxShadow: 'none' }}>
    <h2 style={{ color: Color.Orange.rgb().string(), fontWeight: 100, textAlign: 'center' }}>
      {value}
    </h2>
    <h5 style={{ textAlign: 'center' }}>
      {label}
    </h5>
  </Card.Grid>
)

interface Props {
  workspace: Workspace
  onSelected: (id: number) => void
}

const WorkspaceListItem = ({ workspace, onSelected }: Props) => {
  const {
    id,
    name,
    behavior,
    summary,
    approvals
  } = workspace;

  const onClick = () => onSelected(id);

  const approvalMessage = () => {
    let message = '';
    let color = '';
    
    if(!approvals) {
      message = 'all approvals needed';
      color = 'red';
    } else if(!approvals.infra || !approvals.risk) {
      message = 'partially approved';
      color = 'orange';
    }

    return message === '' ? false : <Tag color={color}>{message}</Tag>;
  }

  return (
    <List.Item>
      <Card
        bordered={true}
        style={{ textAlign: 'center' }}
        onClick={onClick}
        hoverable={true}>
        <h2 style={{ textAlign: 'center' }}>{name}</h2>
        <div style={{ marginTop: 10, marginBottom: 10 }}>
          <Icon style={{ fontSize: 42 }} type={behavior === 'simple' ? 'team' : 'deployment-unit' } />
          <div style={{ fontSize: 12, textTransform: 'uppercase' }}>{behavior} workspace</div>
        </div>
        <h3>{summary}</h3>
        {approvalMessage()}
      </Card>
    </List.Item>
  );
};

export default WorkspaceListItem;