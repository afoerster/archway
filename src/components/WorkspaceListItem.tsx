import { Card, Icon, List, Tag } from 'antd';
import * as React from 'react';
import { Workspace } from '../types/Workspace';

interface Props {
  workspace: Workspace;
  onSelected: (id: number) => void;
}

const WorkspaceListItem = ({ workspace, onSelected }: Props) => {
  const {
    id,
    name,
    behavior,
    summary,
    approvals,
  } = workspace;

  const onClick = () => onSelected(id);

  const approvalMessage = () => {
    let message = '';
    let color = '';

    if (!approvals) {
      message = 'all approvals needed';
      color = 'red';
    } else if (!approvals.infra || !approvals.risk) {
      message = 'partially approved';
      color = 'orange';
    }

    return message === '' ? false : <Tag color={color}>{message}</Tag>;
  };

  return (
    <List.Item>
      <Card
        bordered={true}
        style={{ textAlign: 'center' }}
        onClick={onClick}
        hoverable={true}>
        <h2 style={{ textAlign: 'center' }}>{name}</h2>
        <div style={{ marginTop: 10, marginBottom: 10 }}>
          <Icon style={{ fontSize: 42 }} type={behavior === 'simple' ? 'team' : 'deployment-unit'} />
          <div style={{ fontSize: 12, textTransform: 'uppercase' }}>{behavior} workspace</div>
        </div>
        <h3>{summary}</h3>
        {approvalMessage()}
      </Card>
    </List.Item>
  );
};

export default WorkspaceListItem;
