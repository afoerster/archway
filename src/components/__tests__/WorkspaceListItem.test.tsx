import * as React from 'react';
import { shallow } from 'enzyme';

import WorkspaceListItem from '../WorkspaceListItem';
import { Workspace } from '../../types/Workspace';

describe('WorkspaceListItem', () => {
  it('renders correctly', () => {
    const workspace: Workspace = {
      id: 0,
      name: '',
      summary: '',
      description: '',
      behavior: '',
      requested_date: new Date(0),
      requester: '',
      single_user: false,
      compliance: {
        phi_data: false,
        pci_data: false,
        pii_data: false,
      },
      data: [{
        id: 1,
        name: '',
        location: '',
        size_in_gb: 0,
        consumed_in_gb: 0,
        managing_group: {
          group: {
            common_name: '',
            distinguished_name: '',
            sentry_role: '',
          },
        },
      }],
      processing: [{
        id: 0,
        pool_name: '',
        max_cores: 0,
        max_memory_in_gb: 0,
      }],
      topics: [],
      applications: [],
    };
    const wrapper = shallow(<WorkspaceListItem workspace={workspace} onSelected={() => null} />);
    expect(wrapper).toMatchSnapshot();
  });
});
