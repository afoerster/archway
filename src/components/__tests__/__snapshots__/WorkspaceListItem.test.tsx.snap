// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WorkspaceListItem renders correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <WorkspaceListItem
    onSelected={[Function]}
    workspace={
      Object {
        "applications": Array [],
        "behavior": "",
        "compliance": Object {
          "pci_data": false,
          "phi_data": false,
          "pii_data": false,
        },
        "data": Array [
          Object {
            "consumed_in_gb": 0,
            "location": "",
            "managing_group": Object {
              "group": Object {
                "common_name": "",
                "distinguished_name": "",
                "sentry_role": "",
              },
            },
            "name": "",
            "size_in_gb": 0,
          },
        ],
        "description": "",
        "id": 0,
        "name": "",
        "processing": Array [
          Object {
            "id": 0,
            "max_cores": 0,
            "max_memory_in_gb": 0,
            "pool_name": "",
          },
        ],
        "requested_date": 1970-01-01T00:00:00.000Z,
        "requester": "",
        "single_user": false,
        "summary": "",
        "topics": Array [],
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": <Card
        bodyStyle={
          Object {
            "paddingBottom": 50,
            "paddingTop": 50,
            "textAlign": "center",
          }
        }
        bordered={true}
        hoverable={true}
        onClick={[Function]}
      >
        <ApprovalMessage />
        <Row
          gutter={0}
          type="flex"
        >
          <Col
            span={24}
          >
            <div
              style={
                Object {
                  "fontSize": 26,
                }
              }
            >
              
            </div>
            <div
              style={
                Object {
                  "marginBottom": 10,
                  "marginTop": 10,
                  "textAlign": "center",
                }
              }
            >
              <Icon
                style={
                  Object {
                    "fontSize": 20,
                  }
                }
                type="deployment-unit"
              />
              <div
                style={
                  Object {
                    "fontSize": 12,
                    "textTransform": "uppercase",
                  }
                }
              >
                
                 workspace
              </div>
            </div>
            <p>
              
            </p>
          </Col>
        </Row>
      </Card>,
      "style": Object {
        "paddingLeft": 12,
        "paddingRight": 12,
      },
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "bodyStyle": Object {
          "paddingBottom": 50,
          "paddingTop": 50,
          "textAlign": "center",
        },
        "bordered": true,
        "children": Array [
          <ApprovalMessage />,
          <Row
            gutter={0}
            type="flex"
          >
            <Col
              span={24}
            >
              <div
                style={
                  Object {
                    "fontSize": 26,
                  }
                }
              >
                
              </div>
              <div
                style={
                  Object {
                    "marginBottom": 10,
                    "marginTop": 10,
                    "textAlign": "center",
                  }
                }
              >
                <Icon
                  style={
                    Object {
                      "fontSize": 20,
                    }
                  }
                  type="deployment-unit"
                />
                <div
                  style={
                    Object {
                      "fontSize": 12,
                      "textTransform": "uppercase",
                    }
                  }
                >
                  
                   workspace
                </div>
              </div>
              <p>
                
              </p>
            </Col>
          </Row>,
        ],
        "hoverable": true,
        "onClick": [Function],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <Col
              span={24}
            >
              <div
                style={
                  Object {
                    "fontSize": 26,
                  }
                }
              >
                
              </div>
              <div
                style={
                  Object {
                    "marginBottom": 10,
                    "marginTop": 10,
                    "textAlign": "center",
                  }
                }
              >
                <Icon
                  style={
                    Object {
                      "fontSize": 20,
                    }
                  }
                  type="deployment-unit"
                />
                <div
                  style={
                    Object {
                      "fontSize": 12,
                      "textTransform": "uppercase",
                    }
                  }
                >
                  
                   workspace
                </div>
              </div>
              <p>
                
              </p>
            </Col>,
            "gutter": 0,
            "type": "flex",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": Array [
                <div
                  style={
                    Object {
                      "fontSize": 26,
                    }
                  }
                >
                  
                </div>,
                <div
                  style={
                    Object {
                      "marginBottom": 10,
                      "marginTop": 10,
                      "textAlign": "center",
                    }
                  }
                >
                  <Icon
                    style={
                      Object {
                        "fontSize": 20,
                      }
                    }
                    type="deployment-unit"
                  />
                  <div
                    style={
                      Object {
                        "fontSize": 12,
                        "textTransform": "uppercase",
                      }
                    }
                  >
                    
                     workspace
                  </div>
                </div>,
                <p>
                  
                </p>,
              ],
              "span": 24,
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "",
                  "style": Object {
                    "fontSize": 26,
                  },
                },
                "ref": null,
                "rendered": "",
                "type": "div",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    <Icon
                      style={
                        Object {
                          "fontSize": 20,
                        }
                      }
                      type="deployment-unit"
                    />,
                    <div
                      style={
                        Object {
                          "fontSize": 12,
                          "textTransform": "uppercase",
                        }
                      }
                    >
                      
                       workspace
                    </div>,
                  ],
                  "style": Object {
                    "marginBottom": 10,
                    "marginTop": 10,
                    "textAlign": "center",
                  },
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "function",
                    "props": Object {
                      "style": Object {
                        "fontSize": 20,
                      },
                      "type": "deployment-unit",
                    },
                    "ref": null,
                    "rendered": null,
                    "type": [Function],
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": Array [
                        "",
                        " workspace",
                      ],
                      "style": Object {
                        "fontSize": 12,
                        "textTransform": "uppercase",
                      },
                    },
                    "ref": null,
                    "rendered": Array [
                      "",
                      " workspace",
                    ],
                    "type": "div",
                  },
                ],
                "type": "div",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "",
                },
                "ref": null,
                "rendered": "",
                "type": "p",
              },
            ],
            "type": [Function],
          },
          "type": [Function],
        },
      ],
      "type": [Function],
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <Card
          bodyStyle={
            Object {
              "paddingBottom": 50,
              "paddingTop": 50,
              "textAlign": "center",
            }
          }
          bordered={true}
          hoverable={true}
          onClick={[Function]}
        >
          <ApprovalMessage />
          <Row
            gutter={0}
            type="flex"
          >
            <Col
              span={24}
            >
              <div
                style={
                  Object {
                    "fontSize": 26,
                  }
                }
              >
                
              </div>
              <div
                style={
                  Object {
                    "marginBottom": 10,
                    "marginTop": 10,
                    "textAlign": "center",
                  }
                }
              >
                <Icon
                  style={
                    Object {
                      "fontSize": 20,
                    }
                  }
                  type="deployment-unit"
                />
                <div
                  style={
                    Object {
                      "fontSize": 12,
                      "textTransform": "uppercase",
                    }
                  }
                >
                  
                   workspace
                </div>
              </div>
              <p>
                
              </p>
            </Col>
          </Row>
        </Card>,
        "style": Object {
          "paddingLeft": 12,
          "paddingRight": 12,
        },
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "bodyStyle": Object {
            "paddingBottom": 50,
            "paddingTop": 50,
            "textAlign": "center",
          },
          "bordered": true,
          "children": Array [
            <ApprovalMessage />,
            <Row
              gutter={0}
              type="flex"
            >
              <Col
                span={24}
              >
                <div
                  style={
                    Object {
                      "fontSize": 26,
                    }
                  }
                >
                  
                </div>
                <div
                  style={
                    Object {
                      "marginBottom": 10,
                      "marginTop": 10,
                      "textAlign": "center",
                    }
                  }
                >
                  <Icon
                    style={
                      Object {
                        "fontSize": 20,
                      }
                    }
                    type="deployment-unit"
                  />
                  <div
                    style={
                      Object {
                        "fontSize": 12,
                        "textTransform": "uppercase",
                      }
                    }
                  >
                    
                     workspace
                  </div>
                </div>
                <p>
                  
                </p>
              </Col>
            </Row>,
          ],
          "hoverable": true,
          "onClick": [Function],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": <Col
                span={24}
              >
                <div
                  style={
                    Object {
                      "fontSize": 26,
                    }
                  }
                >
                  
                </div>
                <div
                  style={
                    Object {
                      "marginBottom": 10,
                      "marginTop": 10,
                      "textAlign": "center",
                    }
                  }
                >
                  <Icon
                    style={
                      Object {
                        "fontSize": 20,
                      }
                    }
                    type="deployment-unit"
                  />
                  <div
                    style={
                      Object {
                        "fontSize": 12,
                        "textTransform": "uppercase",
                      }
                    }
                  >
                    
                     workspace
                  </div>
                </div>
                <p>
                  
                </p>
              </Col>,
              "gutter": 0,
              "type": "flex",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": Array [
                  <div
                    style={
                      Object {
                        "fontSize": 26,
                      }
                    }
                  >
                    
                  </div>,
                  <div
                    style={
                      Object {
                        "marginBottom": 10,
                        "marginTop": 10,
                        "textAlign": "center",
                      }
                    }
                  >
                    <Icon
                      style={
                        Object {
                          "fontSize": 20,
                        }
                      }
                      type="deployment-unit"
                    />
                    <div
                      style={
                        Object {
                          "fontSize": 12,
                          "textTransform": "uppercase",
                        }
                      }
                    >
                      
                       workspace
                    </div>
                  </div>,
                  <p>
                    
                  </p>,
                ],
                "span": 24,
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "",
                    "style": Object {
                      "fontSize": 26,
                    },
                  },
                  "ref": null,
                  "rendered": "",
                  "type": "div",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [
                      <Icon
                        style={
                          Object {
                            "fontSize": 20,
                          }
                        }
                        type="deployment-unit"
                      />,
                      <div
                        style={
                          Object {
                            "fontSize": 12,
                            "textTransform": "uppercase",
                          }
                        }
                      >
                        
                         workspace
                      </div>,
                    ],
                    "style": Object {
                      "marginBottom": 10,
                      "marginTop": 10,
                      "textAlign": "center",
                    },
                  },
                  "ref": null,
                  "rendered": Array [
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "function",
                      "props": Object {
                        "style": Object {
                          "fontSize": 20,
                        },
                        "type": "deployment-unit",
                      },
                      "ref": null,
                      "rendered": null,
                      "type": [Function],
                    },
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "children": Array [
                          "",
                          " workspace",
                        ],
                        "style": Object {
                          "fontSize": 12,
                          "textTransform": "uppercase",
                        },
                      },
                      "ref": null,
                      "rendered": Array [
                        "",
                        " workspace",
                      ],
                      "type": "div",
                    },
                  ],
                  "type": "div",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "",
                  },
                  "ref": null,
                  "rendered": "",
                  "type": "p",
                },
              ],
              "type": [Function],
            },
            "type": [Function],
          },
        ],
        "type": [Function],
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
